modules/monitoring/manifests/checks.pp

```diff
@@ -109,6 +109,52 @@
   }
   # END signon
 
+  # START search
+
+  # On average 326 new documents were created per day in 2017, in total.
```

Comment:
I would be more interested in know what the largest change was for a three day period? how often would we have gotten an alert with the below settings?

Reply:
We don't have the historical data in graphite for the actual indexes, so we don't know how many times it would legitimately fire.

With these thresholds there are only 4 days that could have been false positives (if my query works)

```
{"end_day"=>"2017-01-26", "total_3d"=>"1536"}
{"end_day"=>"2017-01-27", "total_3d"=>"1635"}
{"end_day"=>"2017-02-13", "total_3d"=>"4252"}
{"end_day"=>"2017-02-14", "total_3d"=>"4430"}
{"end_day"=>"2017-02-15", "total_3d"=>"4701"}
{"end_day"=>"2017-02-16", "total_3d"=>"5039"}
{"end_day"=>"2017-02-23", "total_3d"=>"2068"}
{"end_day"=>"2017-02-24", "total_3d"=>"2609"}
{"end_day"=>"2017-02-25", "total_3d"=>"2228"}
{"end_day"=>"2017-02-26", "total_3d"=>"1832"}
{"end_day"=>"2017-03-09", "total_3d"=>"1778"}
{"end_day"=>"2017-03-10", "total_3d"=>"1778"}
{"end_day"=>"2017-03-16", "total_3d"=>"1665"}
{"end_day"=>"2017-03-17", "total_3d"=>"1637"}
{"end_day"=>"2017-03-23", "total_3d"=>"1545"}
{"end_day"=>"2017-03-30", "total_3d"=>"1737"}
{"end_day"=>"2017-03-31", "total_3d"=>"1725"}
{"end_day"=>"2017-04-05", "total_3d"=>"1521"}
{"end_day"=>"2017-04-06", "total_3d"=>"1966"}
{"end_day"=>"2017-04-07", "total_3d"=>"1951"}
{"end_day"=>"2017-04-21", "total_3d"=>"1661"}
{"end_day"=>"2017-06-28", "total_3d"=>"2202"}
{"end_day"=>"2017-06-29", "total_3d"=>"5892"}
{"end_day"=>"2017-06-30", "total_3d"=>"5977"}
{"end_day"=>"2017-07-01", "total_3d"=>"5350"}
{"end_day"=>"2017-07-02", "total_3d"=>"4012"}
{"end_day"=>"2017-07-07", "total_3d"=>"2313"}
{"end_day"=>"2017-07-08", "total_3d"=>"2081"}
{"end_day"=>"2017-07-09", "total_3d"=>"1720"}
{"end_day"=>"2017-07-10", "total_3d"=>"1625"}
{"end_day"=>"2017-07-14", "total_3d"=>"1501"}
{"end_day"=>"2017-07-20", "total_3d"=>"1850"}
{"end_day"=>"2017-07-21", "total_3d"=>"1834"}
{"end_day"=>"2017-07-26", "total_3d"=>"1530"}
{"end_day"=>"2017-07-27", "total_3d"=>"1864"}
{"end_day"=>"2017-07-28", "total_3d"=>"1846"}
{"end_day"=>"2017-08-02", "total_3d"=>"1516"}
{"end_day"=>"2017-08-03", "total_3d"=>"1726"}
{"end_day"=>"2017-08-07", "total_3d"=>"16419"}
{"end_day"=>"2017-08-08", "total_3d"=>"16459"}
{"end_day"=>"2017-08-09", "total_3d"=>"16648"}
{"end_day"=>"2017-08-10", "total_3d"=>"17465"}
{"end_day"=>"2017-08-11", "total_3d"=>"1729"}
{"end_day"=>"2017-08-12", "total_3d"=>"1534"}
{"end_day"=>"2017-09-07", "total_3d"=>"1578"}
{"end_day"=>"2017-09-08", "total_3d"=>"1570"}
{"end_day"=>"2017-09-14", "total_3d"=>"1509"}
{"end_day"=>"2017-09-20", "total_3d"=>"1715"}
{"end_day"=>"2017-09-21", "total_3d"=>"2050"}
{"end_day"=>"2017-10-12", "total_3d"=>"1516"}
{"end_day"=>"2017-10-20", "total_3d"=>"1599"}
{"end_day"=>"2017-11-08", "total_3d"=>"3232"}
{"end_day"=>"2017-11-09", "total_3d"=>"3846"}
{"end_day"=>"2017-11-10", "total_3d"=>"4140"}
{"end_day"=>"2017-11-11", "total_3d"=>"3855"}
```

This is counting documents being created rather than published, and doesn't include unpublishing, so it's an overestimate.

Query:
```sql
results = ActiveRecord::Base.connection.execute("
with days as (
  SELECT date_trunc('day', d)::date as day
  FROM generate_series(CURRENT_DATE-365, CURRENT_DATE-1, '1 day'::interval) d ),

counts as (
  select days.day, count(*) as total
  from days left join documents on (documents.created_at::date = days.day)
  group by days.day
),
alert_periods as (
  select
      day as end_day,
      sum(total) over (order by day ROWS BETWEEN 3 PRECEDING AND CURRENT ROW) as total_3d
  from counts

)
select * from alert_periods where total_3d > 1500 order by end_day
"
)
```

It does look like the warnings are a bit low.